### PR TITLE
[INTG-1543] JSON support for exporting inspections

### PR DESCRIPTION
**Pull request by phanikumarpatchigolla**
Description: Adding support for exporting inspections in JSON format.

Changes:
1. Each inspection will be stored in a file named '$AuditID'
2. Parallel processing of fetching and writing inspections is enabled. 

Testing:
More unit tests are being added.

### CODE REVIEW

File: internal/app/inspections/inspections.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	for _, id := range client.SkipIDs {
+		skipIDs[id] = true
+	}
+
+	// Create a buffered channel of length maxGoroutines
+	guard := make(chan struct{}, maxGoRoutines)
+
+	operation := func(row api.Inspection) {
+		inspection, err := client.apiClient.GetInspection(ctx, row.ID)
+		util.Check(err, "Failed to get inspection")
```
Comment (hkf57) [inline internal/app/inspections/inspections.go:84]: Maybe output the ID here? 
```suggestion
		util.Check(err, fmt.Sprintf("Failed to get inspection %s", row.ID))
```

**Review by hkf57 — COMMENTED** [review]
Looks mostly OK but we need to up the test coverage.
Currently the prescribed minimal coverage (which will be changed from codeclimate's side) is 90%.

**Review by atashbahar — COMMENTED** [review]
LGTM but it would be good if @BCook98 has a look as well. He knows the most about this codebase.

File: cmd/iauditor-exporter/cmd/export/export.go [hunk 2]
```diff
@@ -80,6 +94,20 @@ func runSQL(cmd *cobra.Command, args []string) error {
 	return feed.ExportFeeds(viper.GetViper(), apiClient, exporter)
 }
 
+func runInspectionJSON(cmd *cobra.Command, args []string) error {
+
+	exportPath := fmt.Sprintf("%s/json/", viper.GetString("export.path"))
+	os.MkdirAll(exportPath, os.ModePerm)
```
Comment (BCook98) [inline cmd/iauditor-exporter/cmd/export/export.go:100]: This should return on error

File: internal/app/inspections/inspections.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	"github.com/SafetyCulture/iauditor-exporter/internal/app/config"
+	"github.com/SafetyCulture/iauditor-exporter/internal/app/exporter"
+	"github.com/SafetyCulture/iauditor-exporter/internal/app/util"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+const maxGoRoutines = 10
+
+type tFunc func(api.Inspection)
```
Comment (BCook98) [inline internal/app/inspections/inspections.go:18]: What is tFunc?

**Review by BCook98 — CHANGES_REQUESTED** [review]
There is a number of code paths that aren't covered by tests. Can you please look at adding test cases for these, even if they are low level unit tests.

File: internal/app/inspections/inspections.go (hunk 3 above)
Comment (phanikumarpatchigolla) [reply internal/app/inspections/inspections.go]: This is used as an argument for throttling the function.
I have used this variable at multiple places, so defined it as a seperate type.
With the current code it is used at only one place, so removed this.

**Review by BCook98 — APPROVED** [review]

**Pull request merged**